src/Data/Term.hs

```diff
@@ -61,14 +67,28 @@ shift = shift' 0
 instance Show ResolveError where
   show (NotInScope x) = "Variable not in scope: " ++ x
 
+checkCollisionsHelper :: Map String Int -> Term -> Map String Int
+checkCollisionsHelper m (Var _ _) = m
+checkCollisionsHelper m (Abs x t) = let xs = findWithDefault 0 x m in checkCollisionsHelper (Map.insert x (xs + 1) m) t
+checkCollisionsHelper m (App f x) = checkCollisionsHelper (checkCollisionsHelper m f) x
+
+checkCollisions :: Term -> Set String
+checkCollisions t = fromList $ keys $ Map.filter (== 1) (checkCollisionsHelper Map.empty t)
+
+printTerm :: Set String -> Map String (IntMap Int) -> Int -> Term -> String
+printTerm s m d (Var i x) = x ++ if x `member` s then "" else "_" ++ show ((m Map.! x) IntMap.! (d - i))
+printTerm s m d (Abs x t) = let xs = findWithDefault (IntMap.empty) x m 
+  in "\\" ++ (x ++ if x `member` s then "" else "_" ++ show (size xs)) ++ " -> " ++ printTerm s (Map.insert x (IntMap.insert (d + 1) (size xs) xs) m) (d + 1) t
```

Comment:
I think it is totally okay if disambiguation indices do not form a continuous interval, so IMHO you do not have to carry this `Map String (IntMap Int)` around

Reply:
Also, is it okay if I later add the formatter config which sets max line width at 80 characters?